### PR TITLE
README.md formatting

### PR DESCRIPTION
**Pull request by rsanchezsaez**
Add italics formatting to keywords such as *ResearchKit framework*, *iOS*, *Xcode*, etc., to `README.md` and the *ResearchKit Framework Programming Guide*.

Also hard-wrap those two files at column 100.

### CODE REVIEW

**Comment by julientherier** [comment]
It's not a formatting purpose, but as you work on the README file, I'm wondering why "surveys" section comes before "Informed consent". I see the same order in the documentation, or Apple presentations.

For me, it makes more sense to explain the consent first, and then when a user agrees and participates to the study, she gets access to surveys and/or tasks.

**Comment by rsanchezsaez** [comment]
@julientherier: Good point. However, one could argue that *Surveys* is a more integral part of *ResearchKit* than *Consent*, so I guess they prefer to introduce the more important parts first.

Also note that one could use the *Survey* module but not the *Consent* one (e.g., to build a customer satisfaction survey into a non-medical app).

File: docs/Overview/GuideOverview-template.markdown [hunk 1]
```diff
@@ -1,85 +1,75 @@
...
-obtaining consent, or running an active task, everything in
-the ResearchKit framework is a collection of steps (`ORKStep` objects), which
-together form a task (an `ORKTask` object). To present a task,
-attach the task to a task view controller object
-(`ORKTaskViewController`). When the user completes a step in a task,
-the task view controller generates a step result object
-(`ORKStepResult`) that records the start and end time for that step,
-and any results from the step.
- 
+A task in the *ResearchKit* framework can be a simple ordered sequence of steps, or it can be
```
Comment (YuanZhu-apple) [inline docs/Overview/GuideOverview-template.markdown]: Shouldn't use `*` to surround `ResearchKit framework`?

File: README.md [hunk 2]
```diff
@@ -81,48 +88,49 @@ git clone https://github.com/ResearchKit/ResearchKit.git
 Building
 --------
 
-Build the ResearchKit framework by opening `ResearchKit.xcodeproj` and running the
-`ResearchKit` framework target. Optionally, run the unit tests too.
+Build the *ResearchKit framework* by opening `ResearchKit.xcodeproj` and running the *`ResearchKit`
```
Comment (YuanZhu-apple) [inline README.md]: The second `ResearchKit` framework is to use to specify a target. Remove the surrounding `*`?

**Comment by YuanZhu-apple** [comment]
r=me. Except for a few nit picking.

File: README.md [hunk 3]
```diff
@@ -232,19 +237,18 @@ instruction step:
...
 
-The ResearchKit [`ORKCatalog`](samples/ORKCatalog) sample app is a
-good place to start. Find the project in ResearchKit's
-[`samples`](samples) directory. This project includes a list of all
-the types of steps supported by the ResearchKit framework in one tab, and displays a
-browser for the results of the last completed task in the other tab.
+The *ResearchKit* [`ORKCatalog`](samples/ORKCatalog) sample app is a good place to start. Find the
+project in ResearchKit's [`samples`](samples) directory. This project includes a list of all the
+types of steps supported by the *ResearchKit framework* in one tab, and displays a browser for the
+results of the last completed task in the other tab.
```
Comment (umerkhan-apple) [inline README.md]: Did we also want to mention the Charts tab here?

**Comment by umerkhan-apple** [comment]
r=me.

**Comment by rsanchezsaez** [comment]
Comments taken care of. I have also added a *Charts* section to the `README.md` file.

Note that I noticed the `GuideOverview-template.markdown` *italics* (\*word\*) formatting doesn't translate into the output files generated by *headerdoc*. But I left the formatting there anyway in case they add support for it in the future (currently, the `*` symbols seem to be stripped when building the docs).

**Comment by rsanchezsaez** [comment]
I resolved conflicts and merged `master` into this, do you think merging would be appropriate?

I think the formatting and the *Charts section* are nice to have. Let me know if the latter needs rewriting.

File: docs/Overview/GuideOverview-template.markdown [hunk 4]
```diff
@@ -86,16 +77,17 @@ territory in which the app is made available.
...
+Setting the *ResearchKit framework* `ORK_LOG_LEVEL_NONE` macro to `1` completely silences all ResearchKit logs, overriding any other specified log level. Setting `ORK_LOG_LEVEL_DEBUG`, `ORK_LOG_LEVEL_WARNING`, or `ORK_LOG_LEVEL_ERROR` to `1` enables logging at that level and at those of higher seriousness.
 
 If you do not explicitly set a log level, `ORK_LOG_LEVEL_WARNING=1` is used by default.
 
 You have to set any of these preprocessor macros in your ResearchKit subproject, not in your main project. Within *Xcode*, you can do so by setting any of them in the `Preprocessor Macros` list on the `Build Settings` of your `ResearchKit` framework target.
 
 See these resources if you are using ResearchKit through CocoaPods and need to change the log level: [[1]](http://stackoverflow.com/a/30038120/269753) [[2]](http://www.mokacoding.com/blog/cocoapods-and-custom-build-configurations/).
+available.
```
Comment (YuanZhu-apple) [inline docs/Overview/GuideOverview-template.markdown]: Is this line intentional?

Comment (rsanchezsaez) [inline docs/Overview/GuideOverview-template.markdown]: Nope! Removed. I also took the change to fix some minor *headerdoc* warnings that popped up when regenerating the documentation.

**Comment by YuanZhu-apple** [comment]
@rsanchezsaez I might did something in the 1.3 release which caused the conflict in your PR. Can you take some time to resolve it? Thanks!

**Comment by rsanchezsaez** [comment]
@YuanZhu-apple: Done! :)

**Comment by YuanZhu-apple** [comment]
Thanks!

**Pull request merged**